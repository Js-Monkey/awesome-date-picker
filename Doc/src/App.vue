<template>
  <Header/>
  <router-view/>
  {{lang}}
</template>

<script lang="ts">
import {defineComponent, reactive} from 'vue'
import Header from './components/Header.vue'
export default defineComponent({
  name: 'App',
  components: {
    Header
  },
  setup(){
    const target = reactive({lang: 'zh'})
    function toggleLang(){
      target.lang = target.lang === 'zh'? 'en' : 'zh'
    }
    return {
      toggleLang,
      target
    }
  },
  provide() {
    return {
      bus: this
    }
  },
})
</script>

<style>
* {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  overflow-y: auto;
}

#app {

}
</style>
