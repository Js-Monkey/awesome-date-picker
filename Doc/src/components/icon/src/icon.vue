<template>
    <svg class="b-icon" @click="$emit('click',$event)" :style="{fill:`${color}`,fontSize: fontWidth||width}">
        <use :xlink:href="`#icon-${name}`"></use>
    </svg>
</template>

<script lang="ts">
    import '../../../assets/svg/svg'
    import {defineComponent} from 'vue'
    export default defineComponent({
      name: 'b-icon'
    })
</script>

<style scoped lang="scss">
    @keyframes spin {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }

    .b-icon {
        width: 1em;
        height: 1em;

        &.loading {
            animation: spin 1s infinite linear;
        }
    }
</style>
