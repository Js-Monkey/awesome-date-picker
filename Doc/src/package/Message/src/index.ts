import {h, render} from 'vue'
import MessageComponent from './message.vue'
import {DefineComponent} from 'vue'
let timer = null
const queue = []

const Message = function (message: string): void {
  clearQueue()
  const container = document.createElement('div')
  const vm = h(MessageComponent as any, {message}) as any
  render(vm, container)
  vm.component.ctx.toggleVisibility()
  document.body.appendChild(container)
  queue.push(vm)
  timer = setTimeout(() => {
    closeMessage(vm)
  }, 1500)
}

function clearQueue() {
  clearTimeout(timer)
  queue.forEach(vm => closeMessage(vm))
}

function closeMessage(vm) {
  vm.component.ctx.toggleVisibility()

}


export default Message as unknown as DefineComponent
