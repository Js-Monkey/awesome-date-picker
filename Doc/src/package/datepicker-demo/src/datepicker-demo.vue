<template>
  <div class="better-datepicker-wrapper">
    <div class="better-datepicker-header" style="width:298px;text-align:center">
      <svg viewBox="0 0 1024 1024" style="position:absolute;left:30px;width:14px;height:14px">
        <path
          d="M176 513.7l392.73-395.44a32 32 0 0 0-45.41-45.1L108 491.3a32 32 0 0 0 0.16 45.25L523.48 949a32 32 0 1 0 45.1-45.41z"></path>
        <path
          d="M525.23 513.7L918 118.26a32 32 0 1 0-45.41-45.1L457.27 491.3a32 32 0 0 0 0.16 45.25L872.7 949a32 32 0 0 0 45.1-45.41z"></path>
      </svg>
      <svg viewBox="0 0 1024 1024"
           style="position: absolute; left: 50px; width: 14px; height: 14px; display: inline-block;">
        <path
          d="M721.9968 979.0208l47.0528-47.104-419.94752-419.98848 419.94752-419.90144-47.05792-47.04768-419.93216 419.89632h-0.00512l-47.104 47.09888 47.04768 47.04256z"></path>
      </svg>
      <span style="display: none;">2020 - 2029</span><span class="better-datepicker-pointer-cursor "
                                                           style="display: inline-block;">XXX</span><span
      class="better-datepicker-pointer-cursor " style="display: inline-block;">xxxx</span>
      <svg viewBox="0 0 1024 1024" style="position:absolute;right:30px;transform:rotate(180deg);width:14px;height:14px">
        <path
          d="M176 513.7l392.73-395.44a32 32 0 0 0-45.41-45.1L108 491.3a32 32 0 0 0 0.16 45.25L523.48 949a32 32 0 1 0 45.1-45.41z"></path>
        <path
          d="M525.23 513.7L918 118.26a32 32 0 1 0-45.41-45.1L457.27 491.3a32 32 0 0 0 0.16 45.25L872.7 949a32 32 0 0 0 45.1-45.41z"></path>
      </svg>
      <svg viewBox="0 0 1024 1024"
           style="position: absolute; right: 50px; transform: rotate(180deg); width: 14px; height: 14px; display: inline-block;">
        <path
          d="M721.9968 979.0208l47.0528-47.104-419.94752-419.98848 419.94752-419.90144-47.05792-47.04768-419.93216 419.89632h-0.00512l-47.104 47.09888 47.04768 47.04256z"></path>
      </svg>
    </div>
    <table :style="{color: text}" class="better-datepicker-day"
           style="text-align: center; padding: 5px 20px; width: 320px; display: inline-block;">
      <thead>
      <th style="height:40px;width:40px;padding:3px 0">Su</th>
      <th style="height:40px;width:40px;padding:3px 0">Mo</th>
      <th style="height:40px;width:40px;padding:3px 0">Tu</th>
      <th style="height:40px;width:40px;padding:3px 0">We</th>
      <th style="height:40px;width:40px;padding:3px 0">Th</th>
      <th style="height:40px;width:40px;padding:3px 0">Fr</th>
      <th style="height:40px;width:40px;padding:3px 0">Sa</th>
      </thead>
      <tbody>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="pre">
          <div>28</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>1</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>2</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>3</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>4</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>5</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>6</div>
        </td>
      </tr>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="other range-start">
          <div :style="{background: theme}">7</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other inRange">
          <div :style="{background: range}">8</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other inRange">
          <div :style="{background: range}">9</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other inRange">
          <div :style="{background: range}">10</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other inRange">
          <div :style="{background: range}">11</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other inRange">
          <div :style="{background: range}">12</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other range-end">
          <div :style="{background: theme}">13</div>
        </td>
      </tr>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>14</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>15</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>16</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>17</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>18</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>19</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>20</div>
        </td>
      </tr>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>21</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>22</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>23</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" :style="{color: theme}" class="other">
          <div>24</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>25</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>26</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>27</div>
        </td>
      </tr>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>28</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>29</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>30</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="other">
          <div>31</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>1</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>2</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>3</div>
        </td>
      </tr>
      <tr>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>4</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>5</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>6</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>7</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>8</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>9</div>
        </td>
        <td style="height:40px;width:40px;padding:3px 0" class="next">
          <div>10</div>
        </td>
      </tr>
      </tbody>
    </table>
    <table class="better-datepicker-month" style="text-align: center; padding: 5px 20px; width: 320px; display: none;">
      <tbody>
      <tr>
        <td>
          <div>
            <div>Jan</div>
          </div>
        </td>
        <td>
          <div>
            <div>Feb</div>
          </div>
        </td>
        <td>
          <div>
            <div>Mar</div>
          </div>
        </td>
        <td>
          <div>
            <div>Apr</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <div>May</div>
          </div>
        </td>
        <td>
          <div>
            <div>Jun</div>
          </div>
        </td>
        <td>
          <div>
            <div>Jul</div>
          </div>
        </td>
        <td>
          <div>
            <div>Aug</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>
            <div>Sep</div>
          </div>
        </td>
        <td>
          <div>
            <div>Oct</div>
          </div>
        </td>
        <td>
          <div>
            <div>Nov</div>
          </div>
        </td>
        <td>
          <div>
            <div>Dec</div>
          </div>
        </td>
      </tr>
      </tbody>
    </table>
  </div>
</template>

<script lang="ts">
import '../../../assets/svg/svg'
import {defineComponent, toRefs,} from 'vue'

export default defineComponent({
  name: 'datepicker-demo',
  props:{
    cssLists:{
      type:Array
    }
  },
  computed:{
    theme(){
      return this.cssLists[0].value
    },
    range(){
      return this.cssLists[1].value
    },
    text(){
      return this.cssLists[2].value
    }
  }
})
</script>

<style lang="scss" scoped>
@import "./src/assets/date-picker";

.better-datepicker-wrapper {
  position: relative !important;
  display: inline-block;
  max-height:  357px !important;
}

table {
  margin-top: 0 !important;
}
</style>
