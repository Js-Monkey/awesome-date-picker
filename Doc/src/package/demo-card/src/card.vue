<template>
  <div class="demo-card">
    <slot></slot>
    <div v-show="justCode">
      <slot name="code"></slot>
    </div>
    <div v-show="!justCode" class="demo-card-codeBox-wrapper">
      <b-button :class="{active: !isDemo}" @click="closeDemo">JS</b-button>
      <b-button :class="{active: isDemo}" @click="openDemo">Result</b-button>
      <div class="demo-card-codeBox">
        <div v-show="isDemo">
          <slot name="demo"></slot>
        </div>
        <div ref="code" v-show="!isDemo">
          <slot name="code"></slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import '../../../assets/svg/svg'
import {defineComponent, toRefs, ref} from 'vue'

export default defineComponent({
  name: 'demo-card',
  props: {
    name: {
      type: String,
      default: ''
    },
    fontSize: {
      type: [String, Number],
      default: 20
    },
    fill: {
      type: String,
      default: '#ffffff'
    }
  },
  data() {
    return {
      justCode: true
    }
  },
  setup(props) {
    let {fontSize} = toRefs(props)
    const isDemo = ref(false)
    const openDemo = () => isDemo.value = true
    const closeDemo = () => isDemo.value = false
    return {
      fontSize,
      isDemo,
      openDemo,
      closeDemo
    }
  },
  mounted() {
    this.justCode = !this.$slots.demo()[0].children
  }
})
</script>

<style lang="scss">
@import "./Doc/src/assets/style/color";

.demo {
  margin-top: 80px;
  width: 100vw;
  font-family: Futura;

  strong {
    color: #c0392b;
  }

  &-title {
    h1 {
      color: rgb(244, 224, 233);
      text-align: center;
      letter-spacing: 1px;
      color: rgb(244, 224, 233);
    }
  }

  &-card {
    margin-top: 20px;
    margin-bottom: 30px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: #ffb311;

    &-description {
      width: 50vw;
    }

    &-component {
      text-align: center;
    }

    &-codeBox {
      width: 50vw;
      height: 300px;
      border: 1px solid #3498DB;
      overflow-y: auto;
      overflow-x: hidden;
      &-wrapper {
        button {
          color: #3498DB !important;
          border-color: #3498DB !important;
          border-bottom: none !important;
          &:first-child{
            border-right: none;
            border-top-right-radius: 0 !important;
            border-bottom-right-radius: 0 !important;
            border-bottom-left-radius: 0 !important;
          }
          &:nth-child(2){
            border-left-color: white !important;
            border-top-left-radius: 0 !important;
            border-bottom-left-radius: 0 !important;
            border-bottom-right-radius: 0 !important;
          }
          &.active{
            background: #3498DB !important;
            color: #ffffff !important;
          }
        }
        width: 50vw;
        position: relative;
      }
    }
  }

  pre {
    margin: 0;
    code {
      width: calc(100% - 5px);
      border-radius: 7px;
      height: inherit;
      padding: 10px;
      font-size: 16px;
      min-height: 400px;
    }
  }

  .highlight {
    width: 50vw;
    display: inline-block;
  }
}

</style>
