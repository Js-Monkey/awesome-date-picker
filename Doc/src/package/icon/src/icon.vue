<template>
  <svg class="b-icon" @click="$emit('click',$event)" :style="{fontSize,fill}">
    <use :xlink:href="`#icon-${name}`"></use>
  </svg>
</template>

<script lang="ts">
import '../../../assets/svg/svg'
import {defineComponent, toRefs} from 'vue'

export default defineComponent({
  name: 'b-icon',
  props: {
    name: {
      type: String,
      default: ''
    },
    fontSize: {
      type: Number,
      default: 20
    },
    fill: {
      type: String,
      default: '#ffffff'
    }
  },
  setup(props) {
    let {size} = toRefs(props)
    size += 'px'
    return {
      size
    }
  }
})
</script>

<style scoped lang="scss">
@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.b-icon {
  width: 1em;
  height: 1em;

  &.loading {
    animation: spin 1s infinite linear;
  }
}
</style>
