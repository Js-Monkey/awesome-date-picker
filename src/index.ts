import BD from './core'
import Options from './types/options'
import {BetterPickerInstance} from "./types/core"
import {changeOpt} from "./core/util/default-options"
import {pickerLocale} from "./store/modules/options"
import {destroyed} from "./store"

export function createDatePicker(el: HTMLInputElement, options?: Options): BetterPickerInstance | undefined {
  const picker = BD()
  return picker(el, options)
}

export const defaultOptions =  changeOpt

export const locale =  pickerLocale

export const destroy = destroyed
