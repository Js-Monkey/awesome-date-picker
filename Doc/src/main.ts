import {createApp, nextTick} from 'vue'
import App from './App.vue'
import Router from './router/index'
import {useComponent} from "./package"

const app = createApp(App)
useComponent(app)
import highlightJs from 'highlight.js'
import './assets/style/index.scss'
import ColorPicker from 'element-plus/lib/el-color-picker'
import Msg from 'element-plus/lib/el-message'
import './assets/style/el-message.css'
import 'element-plus/lib/theme-chalk/el-color-picker.css'
import {createDatePicker} from '../../src'

app.config.globalProperties.createDatePicker = createDatePicker
app.config.globalProperties.$Msg = Msg

highlightJs.configure({
  languages: ['js', 'html']
})

Router.afterEach( (event) => {
  nextTick(() => document.querySelectorAll('pre code').forEach(block => highlightJs.highlightBlock(block)))
})
app.use(Router)
app.use(ColorPicker)
app.mount('#app')
