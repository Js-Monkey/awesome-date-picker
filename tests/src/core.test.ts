import {createDatePicker} from "../../src"

describe('createDatePicker', () => {
    const reference = document.createElement('input')
    const instance = createDatePicker(reference)

    it('should do nothing if `value` is null', () => {
        expect(instance?.onChange(null as any)).toBeUndefined()
    })
    it('should assign `callback` to `state.onChange` if `value` is a function', () => {
        function callback(date: string) {
            return date
        }

        expect(instance?.onChange(callback)).toBeUndefined()
    })

    it('should destroy binding between `reference` and `DatePicker`', () => {
        expect(instance?.destroyed()).toBeUndefined()
    })

    it('should update `state.options`', () => {
        expect(instance?.update({placement: "bottom"} as any)).toBeUndefined()
    })

    it('should get current date', () => {
        expect(instance?.getCurrentDate()).toBeNull()
        instance?.destroyed()
        expect(instance?.getCurrentDate()).toBeNull()
    })
})
