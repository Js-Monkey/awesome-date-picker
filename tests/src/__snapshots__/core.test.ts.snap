// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDatePicker should match snapshot 1`] = `
Object {
  "clear": [Function],
  "close": [Function],
  "destroyed": [Function],
  "getCurrentDate": [Function],
  "id": 1,
  "onChange": [Function],
  "open": [Function],
  "update": [Function],
}
`;
